### ISSUE TITLE
testTauEmbeddingProducers had ERRORS

### ISSUE THREAD

**Issue by silviodonato** [cms-sw/cmssw#32376]
We are getting an error in the unit test `TauAnalysis/MCEmbeddingTools`  in `CMSSW_11_3_X_2020-12-02-2300`
https://cmssdt.cern.ch/SDT/cgi-bin/logreader/slc7_amd64_gcc900/CMSSW_11_3_X_2020-12-02-2300/unitTestLogs/TauAnalysis/MCEmbeddingTools#/2001-2001

```
----- Begin Fatal Exception 03-Dec-2020 04:18:24 CET-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 274199 lumi: 125 event: 245779397 stream: 0
   [1] Running path 'MINIAODSIMoutput_step'
   [2] Prefetching for module PoolOutputModule/'MINIAODSIMoutput'
   [3] Prefetching for module PATElectronSlimmer/'slimmedLowPtElectrons'
   [4] Prefetching for module PATElectronSelector/'selectedPatLowPtElectrons'
   [5] Prefetching for module PATElectronProducer/'patLowPtElectrons'
   [6] Prefetching for module GsfElectronProducer/'lowPtGsfElectrons'
   [7] Prefetching for module LowPtGsfElectronCoreProducer/'lowPtGsfElectronCores'
   [8] Prefetching for module LowPtGsfElectronSCProducer/'lowPtGsfElectronSuperClusters'
   [9] Prefetching for module PFElecTkProducer/'lowPtGsfElePfGsfTracks'
   [10] Prefetching for module GsfTrackProducer/'lowPtGsfEleGsfTracks'
   [11] Prefetching for module CkfTrackCandidateMaker/'lowPtGsfEleCkfTrackCandidates'
   [12] Calling method for module LowPtGsfElectronSeedProducer/'lowPtGsfElectronSeeds'
Exception Message:
RefCore: A request to resolve a reference to a product of type 'TrajectorySeed' with ProductID '5:571'
can not be satisfied because the product cannot be found.
Probably the branch containing the product is not stored in the input file.
   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.

----- End Fatal Exception -------------------------------------------------
```

It seems related to #31220

**Comment by silviodonato** [comment]
assign reconstruction, xpog

**Comment by perrotta** [comment]
> It seems related to #31220

@bainbrid

**Comment by silviodonato** [comment]
The error is originated from the last step of https://github.com/cms-sw/cmssw/blob/master/TauAnalysis/MCEmbeddingTools/test/runtests.sh#L39 
```
cmsDriver.py MERGE -s PAT --filein file:simulated_and_cleaned.root  --fileout file:merged.root --era Run2_2016_HIPM --data --scenario pp --conditions auto:run2_data --eventcontent  MINIAODSIM --datatier USER --customise TauAnalysis/MCEmbeddingTools/customisers.customiseMerging --customise_commands "process.patTrigger.processName = cms.string('SIMembedding')" -n -1  || die 'Failure during merging step' $?
```

**Comment by bainbrid** [comment]
Hi all,

I took a look at this.

I am able to reproduce the Exception with local tests. 

When inspecting the command [here](https://github.com/cms-sw/cmssw/blob/master/TauAnalysis/MCEmbeddingTools/test/runtests.sh#L39), I realised through edmConfigDump that this PAT step actually also includes large parts of the RECO (somehow), including the full lowPtElectron RECO sequence and the lowPtElectronSeeds module that gives the exception. 

With @slava77 's help, I've tried to decipher the commands and the `--customise TauAnalysis/MCEmbeddingTools/customisers.customiseMerging` argument, which seems to be responsible for the scheduling of RECO modules. 

I've attempted to suppress the execution of lowPtElectron modules by adding [here](https://github.com/cms-sw/cmssw/blob/master/TauAnalysis/MCEmbeddingTools/python/customisers.py#L411) the following two lines:
```
process.merge_step.remove(process.lowPtGsfElectronTask)
process.merge_step.remove(process.gsfTracksOpenConversions)
```
I _think_ this was successful. However, I then saw errors regarding PAT-related lowPtElectron steps, such as the lowPtGsfLinks module. I added further .remove lines such as:
```
process.merge_step.remove(process.lowPtGsfLinks)
process.merge_step.remove(process.slimmedLowPtElectrons)
```
but this was unsuccessful, I suspect because I see several copies of RECO modules in the various intermediate files and it seems that RECO (or parts of it) are run more than once. 

The relevant PAT modules are
https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/PatAlgos/python/producersLayer1/lowPtElectronProducer_cff.py
https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/PatAlgos/python/selectionLayer1/lowPtElectronSelector_cfi.py
https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/PatAlgos/python/slimming/slimmedLowPtElectrons_cff.py

Apologies, but I cannot decipher the details of what the script is trying to achieve.

**Comment by silviodonato** [comment]
see https://github.com/cms-sw/cmssw/pull/32400

**Comment by bainbrid** [comment]
Ah yes, I missed [this](https://github.com/cms-sw/cmssw/pull/32400/files#diff-3f79f9773241e7b131b2b16b438950e5997d068279751887859ee7d2e19bac74R418) line. Thanks.

**Comment by perrotta** [comment]
+1
- Fixed by #32400

**Comment by mariadalfonso** [comment]
+xpog
